src/realm/object-store/impl/deep_change_checker.cpp

```diff
@@ -50,66 +54,110 @@ void DeepChangeChecker::find_related_tables(DeepChangeChecker::RelatedTables& ou
         for (auto& incoming_link : incoming_link_columns) {
             REALM_ASSERT(incoming_link.first);
             auto& links = complete_mapping[incoming_link.first];
-            links.link_columns.push_back(incoming_link.second);
-            links.connected_tables.push_back(cur_table->get_key());
+            links.forward_links.push_back(incoming_link.second);
+            links.forward_tables.push_back(cur_table->get_key());
+
+            auto& backlinks = complete_mapping[cur_table->get_key()];
+            backlinks.backlink_tables.push_back(incoming_link.first);
+            auto opposite_table = group->get_table(incoming_link.first);
+            auto backlink_column = opposite_table->get_opposite_column(incoming_link.second);
```

Comment:
With fbb7361 backlinks are only added when filtered on.

Regarding the collection of mixed: That's not part of part 1 of the scope. We do have several add-ons though. This could be part of one of those.

Reply:
When taking care of https://github.com/realm/realm-core/pull/4646#discussion_r665633561 I noticed that I had to change handling of collections anyway. I added mixed collection with 6178397.